src/common/admin_socket.cc

```diff
@@ -396,17 +400,22 @@ void AdminSocket::do_accept()
     out.claim_append(o);
   }
   uint32_t len = htonl(out.length());
-  int ret = safe_write(connection_fd, &len, sizeof(len));
+  int ret = safe_send(connection_fd, &len, sizeof(len));
   if (ret < 0) {
     lderr(m_cct) << "AdminSocket: error writing response length "
 		 << cpp_strerror(ret) << dendl;
   } else {
-    if (out.write_fd(connection_fd) < 0) {
+    #ifndef _WIN32
```

Comment:
Nit: instead of sprinkling these `#ifdef`s throughout, perhaps this should always invoke `send_fd` when dealing with sockets and then the function itself has the `#ifdef` to use `send` or just invoke the existing `write_fd` method.